batch-models/src/main/resources/schema.sql

```diff
@@ -0,0 +1,126 @@
+-- Create the 'warehouse' database
+CREATE DATABASE warehouse;
+
+-- Connect to the 'warehouse' database
+\c warehouse;
+
+-- Table: user_detail
+CREATE TABLE user_detail( 
+   user_id VARCHAR(255) PRIMARY KEY NOT NULL,
+   mdo_id  VARCHAR(255),
+   full_name VARCHAR(255),
+   designation VARCHAR(255),
+   email VARCHAR(255),
+   phone VARCHAR(255),
+   groups VARCHAR(255),
+   tag VARCHAR(255),
+   user_registration_date VARCHAR(255),
+   roles VARCHAR(255),
+   gender VARCHAR(255),
+   category VARCHAR(255),
+   external_system VARCHAR(255),
+   external_system_id VARCHAR(255),
+   data_last_generated_on VARCHAR(255)
+);
+
+-- Table: org_hierarchy
+CREATE TABLE org_hierarchy(
+   mdo_id VARCHAR(255) PRIMARY KEY NOT NULL,
+   mdo_name  VARCHAR(255),
+   is_cbp_provider VARCHAR(64),
+   ministry VARCHAR(255),
+   department VARCHAR(255),
+   organization VARCHAR(255),
+   data_last_generated_on DATE
+);
+
+-- Table: user_enrolment
+CREATE TABLE user_enrolment(
+   user_id VARCHAR(255) NOT NULL,
+   batch_id VARCHAR(255) NOT NULL,
+   cbp_id  VARCHAR(255)  NOT NULL,
+   cbp_progress_percentage FLOAT,
+   completed_on VARCHAR(255),
+   certificate_generated VARCHAR(255),
+   certificate_generated_on VARCHAR(255),
+   user_rating FLOAT,
+   resource_count_consumed INTEGER,
+   enrolled_on VARCHAR(255),
+   user_consumption_status VARCHAR(255),
+   data_last_generated_on VARCHAR(255),
+   PRIMARY KEY (user_id, cbp_id, batch_id)
+);
+
+-- Table: cbp
+CREATE TABLE cbp(
+   cbp_id VARCHAR(255) PRIMARY KEY NOT NULL,
+   cbp_provider_id VARCHAR(255),
+   cbp_provider_name VARCHAR(255),
+   cbp_name VARCHAR(255),
+   cbp_type VARCHAR(255),
+   batch_id VARCHAR(255),
+   batch_name VARCHAR(255),
+   batch_start_date VARCHAR(255),
+   batch_end_date VARCHAR(255),
+   cbp_duration VARCHAR(255),
+   cbp_rating FLOAT,
+   last_published_on VARCHAR(255),
+   cbp_retired_on VARCHAR(255),
+   cbp_status VARCHAR(255),
+   resource_count INTEGER,
+   total_certificates_issued INTEGER,
+   cbp_substatus VARCHAR(255),
+   data_last_generated_on VARCHAR(255)
+);
+
+-- Table: assessment_detail
+CREATE TABLE assessment_detail(
+   user_id VARCHAR(255)  NOT NULL,
+   cbp_id VARCHAR(255) NOT NULL,
+   assessment_id VARCHAR(255) NOT NULL,
+   assessment_name VARCHAR(255) ,
+   assessment_type VARCHAR(255),
+   assessment_duration VARCHAR(255),
+   time_spent_by_the_user VARCHAR(255),
+   completion_date VARCHAR(255),
```

Comment:
I was under the impression that all the date fields have been moved away from varchar to date datatypes?